app/src/main/java/com/github/se/travelpouch/ui/travel/EditTravelSettings.kt

```diff
@@ -346,17 +354,47 @@ fun EditTravelSettingsScreen(
                     modifier = Modifier.testTag("addUserEmailField"))
                 Button(
                     onClick = {
-                      listTravelViewModel.addUserToTravel(
+                      profileViewModel.getFsUidByEmail(
                           addUserEmail.value,
-                          selectedTravel!!,
-                          { updatedContainer ->
-                            listTravelViewModel.selectTravel(updatedContainer)
-                            Toast.makeText(context, "User added successfully!", Toast.LENGTH_SHORT)
-                                .show()
-                            setExpandedAddUserDialog(false)
+                          onSuccess = { fsUid ->
+                            val isUserAlreadyAdded =
+                                selectedTravel!!.allParticipants.keys.any { it.fsUid == fsUid }
+                            if (fsUid == profileViewModel.profile.value.fsUid) {
+                              Toast.makeText(
+                                      context,
+                                      "Error: You can't invite yourself",
+                                      Toast.LENGTH_SHORT)
+                                  .show()
+                            } else if (isUserAlreadyAdded) {
+                              Toast.makeText(
+                                      context, "Error: User already added", Toast.LENGTH_SHORT)
+                                  .show()
+                            } else if (fsUid != null) {
+                              notificationViewModel.sendNotification(
+                                  Notification(
+                                      listTravelViewModel.getNewUid(),
```

Comment:
This line is problematic. You must not use uids of travels to create a notification. Uids need to be separate